plugins/logentries/logentries_recv.go

```diff
@@ -0,0 +1,196 @@
+package logentries
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/google/go-querystring/query"
+	"github.com/vektra/cypress"
+)
+
+const cAPIRoot = "https://pull.logentries.com"
+
+type LogentriesRecv struct {
+	*http.Client
+	RootURL     string
+	Options     *EventsOptions
+	EventBuffer chan *cypress.Message
+}
+
+type EventsOptions struct {
+	Start  int    `url:"start,omitempty"`
+	End    int    `url:"end,omitempty"`
+	Filter string `url:"filter,omitempty"`
+	Limit  int    `url:"limit,omitempty"`
+}
+
+type EventsResponse struct {
+	Response string `json:"response"`
+	Reason   string `json:"reason"`
+	Events   []*cypress.Message
+}
+
+func NewLogentriesRecv(key, host, log string, options *EventsOptions, bufferSize int) (*LogentriesRecv, error) {
+	root := fmt.Sprintf("%s/%s/hosts/%s/%s/", cAPIRoot, key, host, log)
+
+	url, err := url.Parse(root)
+	if err != nil {
+		return nil, err
+	}
+
+	return &LogentriesRecv{
+		Client:      &http.Client{},
+		RootURL:     url.String(),
+		Options:     options,
+		EventBuffer: make(chan *cypress.Message, bufferSize),
+	}, nil
+}
+
+func (lr *LogentriesRecv) SetDefaultOptions(o *EventsOptions) *EventsOptions {
+	if o.Start == 0 {
+		o.Start = lr.Options.Start
+	}
+	if o.End == 0 {
+		o.End = lr.Options.End
+	}
+	if o.Filter == "" {
+		o.Filter = lr.Options.Filter
+	}
+	if o.Limit == 0 {
+		o.Limit = lr.Options.Limit
+	}
+
+	return o
+}
+
+func (lr *LogentriesRecv) EncodeURL(o *EventsOptions) string {
+	url := lr.RootURL
+
+	v, _ := query.Values(o)
+	if q := v.Encode(); q != "" {
+		url = url + "?" + q
+	}
+
+	return url
+}
+
+func (lr *LogentriesRecv) GetBody(url string) ([]byte, error) {
+	req, err := http.NewRequest("GET", url, nil)
+
+	resp, err := lr.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
```

Comment:
Instead of reading it all, just return `res.Body` and have GetBody return `(io.Reader, error)`.